### ISSUE TITLE
Electra finetuning issues?

### ISSUE THREAD

**Issue by ddofer** [ThilinaRajapakse/simpletransformers#444]
**Describe the bug**
I've trained a number of Electra language models (with 1-2 epochs) using Electra/LM pretraining, then saving the discrimniator. I've done this with different vocab sizes [80, 500, 15000] , and My text data is long sentences without any spaces). 

When I fine-tune the electra discriminator model on a new classification or multiclass task, performance is abysmal. It barely learns and just outputs the majority class. (I've tried training a bag of chars/n-grams, and a keras LSTM model on the tasks, and they performed very well). 

My guess is that there is a bug with how the data is being processed in the fine tuning. 
**To Reproduce**
Dataset is private but shared via email. Long sentences without spaces, all upper cased. 
Train electra language model, then try to fine tune discriminator on a new classification task. 

**Expected behavior**
Better than random results? 

**Desktop (please complete the following information):**
 - Windows 10. 
Latest transformers package (0.31 as of this time)

**Comment by ThilinaRajapakse** [comment]
To be clear, you suspect an issue in the Language Model fine-tuning, right?

**Comment by ddofer** [comment]
I can confirm that building the language model externally gave great results when fine tuning. (As did some other simple baselines such as LSTM or bag of words on the text [long sentences without spaces])
Fine tuning an electra OR language (MLM) model from ST gave terrible/random results, on the finetuning datasets. 
Presumably there's a bug, I suspect in the processing, maybe related to it being texts without spaces?

**Comment by ThilinaRajapakse** [comment]
It may have been fixed after #390. Or is it still not working?

**Comment by ddofer** [comment]
> It may have been fixed after #390. Or is it still not working?

Nope. I tried after that fix, retrained, no change

**Comment by enijkamp** [comment]
@ThilinaRajapakse I and Phil have implemented and verified Electra on the Glue downstream task.

https://github.com/lucidrains/electra-pytorch

The forward pass is based on the replica of the TF forward pass in HF 3.0.2.

The remaining code is written to closely replicate the TF reference code.

That is, data-preprocessing (including ExampleBuilder), learning rate schedule, gradient clipping, etc. might differ.

I believe the two main difference between ST and our code might be:

(1) For the "small" setting, for the generator, there is a discrepancy of the configuration reported in the paper and used in the TF reference code.
(2) Data preprocessing with example builder.

For the TF reference, after 1M updates, the Glue MRPC accuracy is ~87%. Note, there is high variance in these measurements.

For our code, after 200k updates, the Glue MRPC accuracy is ~82%, which might approach ~87% accuracy after 1M updates.

**Comment by ThilinaRajapakse** [comment]
@enijkamp Thanks for the update (and sorry about the slow response)

Just to clarify, you pre-trained your ELECTRA implementation for 200k steps on the OpenWebText dataset and then fine-tined it on MRPC to get ~82% accuracy, right? Was this with the "small" setting?
